tldw_Server_API/app/core/RAG/rag_service/user_personalization_store.py

```
@@ -27,10 +29,43 @@ class Prior:
+    base = _resolve_path(Path("Databases/user_databases"))
+    base.mkdir(parents=True, exist_ok=True)
+    return base
+
+
+def _safe_user_dir(user_id: str) -> Path:
+    base = _get_user_base_dir()
+    candidate = _resolve_path(base / user_id)
+    try:
+        candidate.relative_to(base)
+    except ValueError as exc:
+        raise ValueError("Unsafe user_id path for personalization store") from exc
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)